### ISSUE TITLE
Plugin should use 'subscription-manager version' to determine whether the VM is already registered

### ISSUE THREAD

**Issue by hferentschik** [projectatomic/adb-vagrant-registration#73]
If it is registered, there is no need to call registration again.

**Comment by pvalena** [comment]
If you set .force = false, it won't register again. The default behavior for now is to re-register if already registered to ensure the subscription is valid.

**Comment by hferentschik** [comment]
{quote}
If you set .force = false, it won't register again. 
{quote}
I want to improve the user experience w/o adding/using additional configuration in the _Vagrantfile_

{quote}
The default behavior for now is to re-register if already registered to ensure the subscription is valid.
{quote}
The problem is that registration takes a very long time. If we can find a faster way to verify that a VM is already registered (for example by using `subscription-manager version` which seems faster`), we will improve the overall user experience.

**Comment by pvalena** [comment]
Yes, i think there can be some general enhancement in way we work with subscription-manager. I will check with subscription-manager maintainers.


>  If we can find a faster way to verify that a VM is already registered 

It will not affect current (default) behavior, as it will register anyway.

**Comment by hferentschik** [comment]
> It will not affect current (default) behavior, as it will register anyway.

Maybe it does not have to. We introduce a validity re-check interval. Just  as an example. Bottom line, right now subscription is very slow, probably one of the biggest time consumers in a 'vagrant up'. Also, for most users, there is no reason why one would un-register and register on each up/halt cycle. 

There are already several options which control the registration behavior, but I believe they have the wrong default. The majority of users will have a single ADB/CDK box which needs to be registered once. We should optimize for this use case. If someone really want to have multiple setups and swtich between them, he can use _config.registration.unregister_on_halt = true_ to make sure his single license gets released.

**Comment by hferentschik** [comment]
See also https://github.com/projectatomic/adb-vagrant-registration/issues/72
